autogluon/searcher/searcher.py

```diff
@@ -155,25 +156,19 @@ def get_config(self, **kwargs):
         -------
         A new configuration that is valid.
         """
-        if len(self._results) == 0: # no hyperparams have been tried yet, first try default config
+        if not self._results:  # no hyperparams have been tried yet, first try default config
             new_config = self.configspace.get_default_configuration().get_dictionary()
         else:
             new_config = self.configspace.sample_configuration().get_dictionary()
         with self.LOCK:
             num_tries = 1
             while pickle.dumps(new_config) in self._results.keys():
                 assert num_tries <= self.MAX_RETRIES, \
-                    "Cannot find new config in BaseSearcher, even after {} trials".format(
-                        self.MAX_RETRIES)
+                    f"Cannot find new config in BaseSearcher, even after {self.MAX_RETRIES} trials"
                 new_config = self.configspace.sample_configuration().get_dictionary()
                 num_tries += 1
-            self._results[pickle.dumps(new_config)] = 0
+            self._results[pickle.dumps(new_config)] = float("-inf")
```

Comment:
Please pay attention here. When we are requesting new config from random searcher, it    already puts the config in the `self._results` dict with value `0`. Whenever reward is coming and it is negative, it will be not updated in results dict due to line 62: `self._results[config_pkl] = max(reward, old_reward)`. So I changed it to infinity to be able to update dict in any case.

Reply:
Is there an example eval metric which causes defective behavior without this change during HPO?

Reply:
yep, for example `r2`, which can be negative.

Reply:
I believe it also fixes logloss because for Tabular we flip the scores such that 0 is the max value and thus you can get scores like -0.25.